### ISSUE TITLE
Use multiple outputs for extras, remove html5lib?

### ISSUE THREAD

**Issue by bollwyvl** [conda-forge/rdflib-feedstock#25]
### Comment:

It would be nice to reflect the other `[extras]` as conda packages:

```
[tool.poetry.extras]
berkeleydb = ["berkeleydb"]
networkx = ["networkx"]
html = ["html5lib"]
lxml = ["lxml"]
```

These would take the form of the relatively-unlikely-to-conflict `rdflib-with-berkeleydb`, built with  [multiple outputs](https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#outputs-section). Rounding this out would be an `rdflib-with-all` package that included all known extras.

As part of this, `html5lib` now kind of in maintenance mode, should probably be dropped from the main package's dependencies.

Another knock-on of this is being able to create an `_rdflib_tests_` package, depending on all of the subpackaged, which a user is very unlikely to install accidentally. This way none of the user-facing packages would include the full test assets.

**Comment by bollwyvl** [comment]
Not looking to make any more work for folks! Just trying to make it easier to get/test the packages efficiently.

**Comment by aucampia** [comment]
I will take a more detailed look when I can, if we can integrate it with our repo then I  think it will be great.

**Comment by bollwyvl** [comment]
> integrate it with our repo

Welp... the `conda-forge` bot is _pretty_ good, and usually we only need to check the dependencies upstream and click a button. When big changes land, like a different build backend, or a different organization of the `sdist`, it takes a little more manual intervention.

I haven't yet seen a good pattern for an upstream _truly_ automating this: basically, to publish to (and be depended on by other packages in) `conda-forge`, the CI needs to run on _this_ repo, from PR on a personal fork. Even adding `conda-build` (or even `conda-mambabuild`) to "normal" CI slows it _way_ down.

Theoretically, one could build some automation at the tail end of an upstream `sdist` release to PyPI that created a PR here (from an `@rdflib`-owned fork), which would then require a single button click here (which we don't mind doing), provided the linters/tests pass... the `automerge` stuff is again reserved for infrastructure that is run by `conda-forge` and handles nasty things like compiler migrations, etc.

There is some automation that is better at looking at dependencies like [grayskull](https://github.com/conda/grayskull), but it's relatively conservative, and doesn't do things like run test suites with coverage... while this is not _required_ in `conda-forge`, I'm personally handling a _lot_ of feedstocks, and the up-front effort of getting the test with coverage has been really effective at keeping this manageable: if it's green, I know I can Press The Button and start using a conda-forge release in production.

**Comment by bollwyvl** [comment]
Of course, we're _always_ interested in upstream maintainers joining the team: feel free to make an issue here and we can get you added (or you can use the magic "bot command" issue, which cleans things up a bit further).

**Comment by bollwyvl** [comment]
Please see the concrete implementation of this on https://github.com/conda-forge/rdflib-feedstock/pull/27.

Of note, i couldn't yet add `rdflib-with-berkeleydb`, as there is no modern build for it yet.

**Comment by bollwyvl** [comment]
Implemented in #27
